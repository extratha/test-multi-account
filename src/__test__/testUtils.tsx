import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
