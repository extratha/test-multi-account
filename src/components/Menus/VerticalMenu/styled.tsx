import { Stack, styled } from "@mui/material";

export const VerticalMenuContainer = styled(Stack)(({ theme }) => ({
  backgroundColor: theme.palette.background.paper,
  maxWidth: "330px",
  minWidth: "330px",
  height: "100vh",
  padding: "1.5rem 1.5rem 1rem",
  overflowY: "scroll",
}));

export const ProfileInfoBox = styled(Stack)({
  background: "linear-gradient(94.13deg, #0059D6 0%, #028DF5 50%, #00CBDD 100%)",
  borderRadius: "24px",
  minHeight: "260px",
  padding: "18px",
  position: "relative",
});
