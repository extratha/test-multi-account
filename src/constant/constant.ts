export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};
