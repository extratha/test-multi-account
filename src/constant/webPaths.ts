import SetNewPasswordPage from "@/app/[locale]/set-new-password/page";

export const webPaths = {
  landing: '/landing',
  home: '/home',
  forgetPassword: '/forget-password',
  setNewPassword: '/set-new-password',
};
