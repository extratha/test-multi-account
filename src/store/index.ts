export * from './modal';
export { default as usePageLoadingStore } from './usePageLoadingStore';
export { default as useUserProfileStore } from './useUserProfileStore';
