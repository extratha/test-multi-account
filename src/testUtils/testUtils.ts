import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
