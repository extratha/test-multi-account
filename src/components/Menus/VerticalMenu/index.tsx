"use client";
import {
  IconCreditCoin,
  IconMenuItem,
  IconMenuItemActive,
  IconPerson,
  IconSparkle,
  ImagePlaygroundLogoColor,
  ImagePlaygroundLogoOverlay,
} from "@/assets";
import { CUSTOM_COLORS } from "@/config/config-mui/theme/colors";
import { MenuItem, aiMenuList, iconStyles, settingMenuList } from "@/constant/menu";
import { webPaths } from "@/constant/webPaths";
import { useUserProfileStore } from "@/store";
import { KeyboardArrowDown, KeyboardArrowUp } from "@mui/icons-material";
import { Collapse, Divider, List, ListItem, Stack, Typography, useTheme } from "@mui/material";
import { deleteCookie } from "cookies-next";
import { useTranslations } from "next-intl";
import { useParams, usePathname, useRouter } from "next/navigation";
import { useEffect, useState } from "react";
import { ProfileInfoBox, VerticalMenuContainer } from "./styled";
const VerticalMenu = () => {
  const t = useTranslations("Common");
  const [isMenuExpand, setIsMenuExpand] = useState<number[]>([]);
  const { data, resetUserProfile } = useUserProfileStore();
  const theme = useTheme();
  const pathname = usePathname();
  const { locale } = useParams();
  const router = useRouter();
  const [isShowMenu, setIsShowMenu] = useState<boolean>(false);
  const onClickExpandMenu = (menuIndex: number) => {
    if (isMenuExpand.some((index) => index === menuIndex)) {
      setIsMenuExpand(isMenuExpand.filter((index) => index !== menuIndex));
    } else {
      setIsMenuExpand([...isMenuExpand, menuIndex]);
    }
  };
  const excludePath = [
    webPaths.login,
    webPaths.forgetPassword,
    webPaths.setNewPassword,
    webPaths.termsAndCons,
    webPaths.aiInterpret.tryInputData,
    webPaths.aiInterpret.aiInterpretResult,
  ];

  useEffect(() => {
    const currentPath = pathname.split(`/${locale}`)[1];
    setIsShowMenu(!excludePath.includes(currentPath));
  }, [pathname]);
  const handleClickHeadMenu = (menu: MenuItem) => {
    if (menu?.title?.includes(t("menu.logout"))) {
      deleteCookie("accessToken");
      deleteCookie("refreshToken");
      deleteCookie("passwordChanged");
      resetUserProfile();
      router.replace(webPaths.login);
      return;
    }
    if (menu.path) {
      router.push(menu.path);
    }
  };
  const handleClickSubmenu = (subMenu: MenuItem) => {
    if (subMenu.path) {
      router.push(subMenu.path);
    }
  };
  const MenuRenderer = (meuList: MenuItem[]) => {
    return (
      <List>
        {meuList.map((menu, index) => {
          return (
            <ListItem key={index} sx={{ padding: "16px 0" }}>
              <Stack width="100%" sx={{ cursor: "pointer" }}>
                <Stack direction="row" width="100%" onClick={() => handleClickHeadMenu(menu)}>
                  <Stack flex={1} margin="auto 16px auto 0 " maxWidth={"24px"} justifyContent={"center"}>
                    {menu.path && pathname.includes(menu.path) ? (
                      <>{menu?.activeIcon || <IconMenuItemActive style={iconStyles} />}</>
                    ) : (
                      <>{menu?.icon || <IconMenuItem style={iconStyles} />}</>
                    )}
                  </Stack>
                  <Stack justifySelf={"start"}>
                    <Typography variant="titleMedium">{menu.title}</Typography>
                  </Stack>
                  {menu.children && menu.children.length > 0 ? (
                    <Stack
                      sx={{ marginLeft: "auto" }}
                      data-testid="icon-expand"
                      onClick={() => onClickExpandMenu(index)}
                    >
                      <>
                        {isMenuExpand.some((number) => number === index) ? (
                          <KeyboardArrowUp />
                        ) : (
                          <KeyboardArrowDown></KeyboardArrowDown>
                        )}
                      </>
                    </Stack>
                  ) : null}
                </Stack>
                <Collapse
                  in={isMenuExpand.some((number) => number === index)}
                  data-testid={`menu-group-collapse-${index}`}
                >
                  <List>
                    {menu.children?.map((childrenItem, childrenIndex) => {
                      return (
                        <ListItem
                          key={childrenIndex}
                          sx={{
                            padding: "16px 40px",
                            color:
                              childrenItem.path && pathname.includes(childrenItem.path) ? "primary.lighter" : "inherit",
                          }}
                          onClick={() => handleClickSubmenu(childrenItem)}
                        >
                          <Typography variant="bodyLarge">{childrenItem.title}</Typography>
                        </ListItem>
                      );
                    })}
                  </List>
                </Collapse>
              </Stack>
            </ListItem>
          );
        })}
      </List>
    );
  };
  return (
    <>
      {isShowMenu ? (
        <VerticalMenuContainer data-testid="vertical-menu-container">
          <ImagePlaygroundLogoColor style={{ margin: "0 0 1.5rem" }} />
          <Typography variant="bodyLargeSemiBold">{t("menu.aiMenus")}</Typography>
          {MenuRenderer(aiMenuList(t))}
          <Divider style={{ marginBottom: "20px" }}></Divider>
          <Typography variant="bodyLargeSemiBold">{t("menu.settingMenus")}</Typography>
          {MenuRenderer(settingMenuList(t))}
          <Divider style={{ margin: "auto 0 20px" }}></Divider>

          <ProfileInfoBox>
            <Stack height="100%">
              <Stack direction="row">
                <IconSparkle style={{ width: 20, height: 20 }} />
                <Stack
                  direction="row"
                  sx={{
                    background: theme.palette.text.primary,
                    borderRadius: "40px",
                    margin: "0 0 0 auto",
                    padding: "8px 16px",
                  }}
                >
                  <IconPerson style={{ width: 24, height: 24, margin: "auto 10px auto 0" }} />
                  <Typography variant="bodyLargeSemiBold" color={theme.palette.background.paper}>
                    {t("roles.admin")}
                  </Typography>
                </Stack>
              </Stack>
              <Stack sx={{ height: "100%", color: theme.palette.background.paper, margin: "16px 0 0 0" }}>
                <Typography variant="bodyLargeSemiBold"> Admin Officer </Typography>
                <Typography variant="bodyLarge"> {data.email} </Typography>

                <Typography sx={{ margin: "auto 0 8px 0" }} variant="bodyLargeSemiBold">
                  เครดิตของฉัน{" "}
                </Typography>
                <Stack direction="row">
                  <IconCreditCoin style={{ width: "24px", height: "24px", margin: "0 1rem 0 0" }} />
                  <Typography variant="titleLargeSemiBold" color={CUSTOM_COLORS.coin}>
                    ใช้ได้ไม่จำกัด
                  </Typography>
                </Stack>
              </Stack>
            </Stack>
            <ImagePlaygroundLogoOverlay
              style={{
                position: "absolute",
                right: 0,
                bottom: 0,
                width: "140px",
                height: "140px",
              }}
            />
          </ProfileInfoBox>
        </VerticalMenuContainer>
      ) : null}
    </>
  );
};

export default VerticalMenu;
