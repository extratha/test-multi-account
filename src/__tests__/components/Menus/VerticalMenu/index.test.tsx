/* eslint-disable testing-library/no-wait-for-multiple-assertions */
/* eslint-disable jest/no-conditional-expect */
/* eslint-disable testing-library/no-wait-for-side-effects */
/* eslint-disable testing-library/no-unnecessary-act */
// TODO: remove disable

import VerticalMenu from "@/components/Menus/VerticalMenu";
import { webPaths } from "@/constant/webPaths";
import { useUserProfileStore } from "@/store";
import { deleteCookie } from "cookies-next";
import mockRouter from "next-router-mock";
import { act, fireEvent, render, screen, waitFor } from "../../../testUtils";

mockRouter.setCurrentUrl("/th/home");

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useRouter: jest.fn(),
  usePathname: jest.fn().mockReturnValue("/th/home"),
  useParams: jest.fn(() => ({ locale: "th" })),
}));

jest.mock("../../../../store", () => ({
  useUserProfileStore: jest.fn(),
}));

jest.mock("cookies-next", () => ({
  deleteCookie: jest.fn(),
}));
const mockRouterReplace = jest.fn();
const mockResetUserProfile = jest.fn();

describe("ProjectCoverLayout", () => {
  beforeEach(() => {
    (useUserProfileStore as any).mockReturnValue({
      data: { email: "admin@example.com" },
      resetUserProfile: mockResetUserProfile,
    });

    // TODO: refactor
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const useRouter = require("next/navigation").useRouter;
    useRouter.mockReturnValue({
      push: jest.fn(),
      replace: mockRouterReplace,
    });
  });

  it("renders menu items correctly", async () => {
    await act(async () => render(<VerticalMenu />));

    expect(screen.getByText(/เครื่องมือวิเคราะห์ข้อมูลด้วย AI/)).toBeInTheDocument();
    expect(screen.getByText(/การตั้งค่าและอื่นๆ/)).toBeInTheDocument();
  });

  it("handles menu expand and collapse", async () => {
    await act(async () => render(<VerticalMenu />));

    const firstMenu = screen.getByText(/AI Interpret/).closest("li");
    if (firstMenu) {
      const expandIcon = screen.getByTestId("icon-expand");

      if (expandIcon) {
        await act(async () => {
          fireEvent.click(expandIcon);
          await waitFor(() => {
            expect(screen.getByText(/ทดลองใช้ข้อมูลตัวอย่าง/)).toBeInTheDocument();
          });
        });

        await act(async () => {
          const groupCollapse = screen.getByTestId(`menu-group-collapse-${1}`);
          await waitFor(() => {
            fireEvent.click(expandIcon);
          });
          await waitFor(async () => {
            expect(groupCollapse).toHaveStyle("height: 0px");
          });
        });
      }
    }
  });

  it("handles logout correctly", async () => {
    await act(async () => render(<VerticalMenu />));
    fireEvent.click(screen.getByText(/ออกจากระบบ/));

    await waitFor(() => {
      expect(deleteCookie).toHaveBeenCalledWith("accessToken");
      expect(deleteCookie).toHaveBeenCalledWith("refreshToken");
      expect(deleteCookie).toHaveBeenCalledWith("passwordChanged");
      expect(mockResetUserProfile).toHaveBeenCalled();
      expect(mockRouterReplace).toHaveBeenCalledWith(webPaths.login);
    });
  });

  it("hides menu based on pathname change", async () => {
    render(<VerticalMenu />);

    screen.getByTestId("vertical-menu-container");
    mockRouter.setCurrentUrl(`/th${webPaths.login}`);
  });

  it("displays user profile info correctly", async () => {
    await act(async () => render(<VerticalMenu />));

    expect(screen.getByText("admin@example.com")).toBeInTheDocument();
    expect(screen.getByText("ใช้ได้ไม่จำกัด")).toBeInTheDocument();
  });

  it("to match snap", async () => {
    const { asFragment } = await act(async () => render(<VerticalMenu></VerticalMenu>));
    expect(asFragment()).toMatchSnapshot();
  });
});
