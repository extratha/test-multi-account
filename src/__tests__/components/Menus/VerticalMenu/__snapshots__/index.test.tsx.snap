// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCoverLayout to match snap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  max-width: 330px;
  min-width: 330px;
  height: 100vh;
  padding: 1.5rem 1.5rem 1rem;
  overflow-y: scroll;
}

.emotion-1 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
}

.emotion-2 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding: 16px 0;
}

.emotion-3.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-3.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-3.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-3.Mui-disabled {
  opacity: 0.38;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  cursor: pointer;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: auto 16px auto 0;
  max-width: 24px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  justify-self: start;
}

.emotion-8 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.15px;
  font-size: 16px;
  font-family: NotoSansThai-Medium,sans-serif;
}

.emotion-9 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-11 {
  width: 100%;
}

.emotion-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: auto;
}

.emotion-20 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding: 16px 40px;
  color: inherit;
}

.emotion-25.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-25.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-25.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-25.Mui-disabled {
  opacity: 0.38;
}

.emotion-26 {
  margin: 0;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-Regular,sans-serif;
}

.emotion-31 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-45 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: linear-gradient(94.13deg, #0059D6 0%, #028DF5 50%, #00CBDD 100%);
  border-radius: 24px;
  min-height: 260px;
  padding: 18px;
  position: relative;
}

.emotion-46 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.emotion-47 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-48 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: rgba(0, 0, 0, 0.87);
  border-radius: 40px;
  margin: 0 0 0 auto;
  padding: 8px 16px;
}

.emotion-49 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  color: #fff;
}

.emotion-50 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  color: #fff;
  margin: 16px 0 0 0;
}

.emotion-53 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  margin: auto 0 8px 0;
}

.emotion-55 {
  margin: 0;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  font-size: 20px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  color: #FFD54F;
}

<div
    class="MuiStack-root emotion-0"
    data-testid="vertical-menu-container"
  >
    <img
      alt=""
      src="/img.jpg"
      style="margin: 0px 0px 1.5rem;"
    />
    <span
      class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-1"
    >
      เครื่องมือวิเคราะห์ข้อมูลด้วย AI
    </span>
    <ul
      class="MuiList-root MuiList-padding emotion-2"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-3"
      >
        <div
          class="MuiStack-root emotion-4"
        >
          <div
            class="MuiStack-root emotion-5"
          >
            <div
              class="MuiStack-root emotion-6"
            >
              <img
                alt="icon-menu-item-active"
                src="/img.jpg"
                style="width: 24px; height: 24px;"
              />
            </div>
            <div
              class="MuiStack-root emotion-7"
            >
              <h3
                class="MuiTypography-root MuiTypography-titleMedium emotion-8"
              >
                หน้าหลัก
              </h3>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-9"
            data-testid="menu-group-collapse-0"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical emotion-10"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-11"
              >
                <ul
                  class="MuiList-root MuiList-padding emotion-2"
                />
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-3"
      >
        <div
          class="MuiStack-root emotion-4"
        >
          <div
            class="MuiStack-root emotion-5"
          >
            <div
              class="MuiStack-root emotion-6"
            >
              <img
                alt="icon-menu-item"
                src="/img.jpg"
                style="width: 24px; height: 24px;"
              />
            </div>
            <div
              class="MuiStack-root emotion-7"
            >
              <h3
                class="MuiTypography-root MuiTypography-titleMedium emotion-8"
              >
                AI Interpret
              </h3>
            </div>
            <div
              class="MuiStack-root emotion-19"
              data-testid="icon-expand"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-20"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-9"
            data-testid="menu-group-collapse-1"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical emotion-10"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-11"
              >
                <ul
                  class="MuiList-root MuiList-padding emotion-2"
                >
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-25"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-bodyLarge emotion-26"
                    >
                      ทดลองใช้ข้อมูลตัวอย่าง
                    </span>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-25"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-bodyLarge emotion-26"
                    >
                      ทดลองใส่ข้อมูลเอง
                    </span>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-25"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-bodyLarge emotion-26"
                    >
                      ใช้ข้อมูลจากโรงพยาบาล
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth emotion-31"
      style="margin-bottom: 20px;"
    />
    <span
      class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-1"
    >
      การตั้งค่าและอื่นๆ
    </span>
    <ul
      class="MuiList-root MuiList-padding emotion-2"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-3"
      >
        <div
          class="MuiStack-root emotion-4"
        >
          <div
            class="MuiStack-root emotion-5"
          >
            <div
              class="MuiStack-root emotion-6"
            >
              <img
                alt="icon-menu-item"
                src="/img.jpg"
                style="width: 24px; height: 24px;"
              />
            </div>
            <div
              class="MuiStack-root emotion-7"
            >
              <h3
                class="MuiTypography-root MuiTypography-titleMedium emotion-8"
              >
                ออกจากระบบ
              </h3>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-9"
            data-testid="menu-group-collapse-0"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical emotion-10"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-11"
              >
                <ul
                  class="MuiList-root MuiList-padding emotion-2"
                />
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth emotion-31"
      style="margin: auto 0px 20px;"
    />
    <div
      class="MuiStack-root emotion-45"
    >
      <div
        class="MuiStack-root emotion-46"
      >
        <div
          class="MuiStack-root emotion-47"
        >
          <img
            alt=""
            src="/img.jpg"
            style="width: 20px; height: 20px;"
          />
          <div
            class="MuiStack-root emotion-48"
          >
            <img
              alt=""
              src="/img.jpg"
              style="width: 24px; height: 24px; margin: auto 10px auto 0px;"
            />
            <span
              class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-49"
            >
              แอดมิน
            </span>
          </div>
        </div>
        <div
          class="MuiStack-root emotion-50"
        >
          <span
            class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-1"
          >
             Admin Officer 
          </span>
          <span
            class="MuiTypography-root MuiTypography-bodyLarge emotion-26"
          >
             admin@example.com 
          </span>
          <span
            class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-53"
          >
            เครดิตของฉัน 
          </span>
          <div
            class="MuiStack-root emotion-47"
          >
            <img
              alt=""
              src="/img.jpg"
              style="width: 24px; height: 24px; margin: 0px 16px 0px 0px;"
            />
            <h3
              class="MuiTypography-root MuiTypography-titleLargeSemiBold emotion-55"
            >
              ใช้ได้ไม่จำกัด
            </h3>
          </div>
        </div>
      </div>
      <img
        alt=""
        src="/img.jpg"
        style="position: absolute; right: 0px; bottom: 0px; width: 140px; height: 140px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
